script/bench.rb

```diff
@@ -207,32 +206,22 @@ def bench(path, name, headers)
     run("bundle exec rake assets:precompile")
   end
 
-  pid =
-    if @unicorn
-      ENV["UNICORN_PORT"] = @port.to_s
-      ENV["UNICORN_WORKERS"] = @unicorn_workers.to_s
-      FileUtils.mkdir_p(File.join("tmp", "pids"))
-      unicorn_pid = spawn("bundle exec unicorn -c config/unicorn.conf.rb")
-
-      while (
-              unicorn_master_pid =
-                `ps aux | grep "unicorn master" | grep -v "grep" | awk '{print $2}'`.strip.to_i
-            ) == 0
-        sleep 1
-      end
-
-      while `ps -f --ppid #{unicorn_master_pid} | grep worker | awk '{ print $2 }'`.split("\n")
-              .map(&:to_i)
-              .size != @unicorn_workers.to_i
-        sleep 1
-      end
-
-      unicorn_pid
-    else
-      spawn("bundle exec puma -p #{@port} -e production")
-    end
+  ENV["UNICORN_PORT"] = @port.to_s
+  ENV["UNICORN_WORKERS"] = @unicorn_workers.to_s
+
+  FileUtils.mkdir_p(File.join("tmp", "pids"))
+  pid = spawn("bundle exec unicorn -c config/unicorn.conf.rb")
+
+  while (
+          unicorn_master_pid =
+            `ps aux | grep "unicorn master" | grep -v "grep" | awk '{print $2}'`.strip.to_i
+        ) == 0
+    sleep 1
+  end
 
-  sleep 1 while port_available? @port
```

Comment:
This fails on macOS but we don't really need it when starting Unicorn because we are already waiting for all the unicorn workers to be up before proceeding.